bin/update-third-party-notices.sh

```diff
@@ -8,15 +8,81 @@
 
 STAGED_FILES=$(git diff-index --cached --name-only HEAD)
 
+PKG_HAS_CHANGED=0
 for FILE in $STAGED_FILES
 do
-  if [ $FILE == "package.json" ]; then
-    RUN_THIRD_PARTY=1
+  if [ ${FILE} == "package.json" ]; then
+    PKG_HAS_CHANGED=1
     break
   fi
 done
 
-if [ -n "$RUN_THIRD_PARTY" ]; then
+if [ ${PKG_HAS_CHANGED} -eq 0 ]; then
+  exit 0
```

Comment:
can we add an echo here so we now it's a no-op. same for at the end when it doesn't updating third party deps?